specs/chain/hs/src/Cardano/Spec/Chain/STS/Block.hs

```diff
@@ -17,32 +18,41 @@ import Ledger.Core
 import Ledger.Delegation
 import Ledger.Signatures
 
-data BlockHeader
-  = BlockHeader
-  { _prevHHash :: !Hash
-    -- ^ Hash of the previous block header, or 'genesisHash' in case of the
-    -- first block in a chain.
-  , _bSlot :: !Slot
-    -- ^ Absolute slot for which the block was generated.
-  , _bIssuer :: !VKey
-    -- ^ Block issuer.
-  , _bSig :: !(Sig VKey)
-    -- ^ Signature of the block by its issuer.
+-- TODO(md): At the moment there is no description of the type in the
+-- LaTeX specification.
+data BHToSign deriving (Eq, Show, Generic)
 
+data BlockHeader
+  = MkBlockHeader
+  {
+    -- | Hash of the previous block header, or 'genesisHash' in case of
+    -- the first block in a chain.
+    _bhPrevHash :: Hash
+    -- | Hash of the block.
+  , _bhHash     :: Hash
+    -- | Absolute slot for which the block was generated.
+  , _bhSlot :: Slot
+    -- | Block issuer.
+  , _bhIssuer   :: VKey
+    -- | Part of the block header which must be signed.
+  , _bhToSign   :: BHToSign
```

Comment:
I think this shouldn't be a specific field but a lens, since it's not extra data in the header, it's some collection of the data already there. We could initialise it as being everything apart from the signature.

Reply:
As I mentioned it earlier, there is no explanation of this field in the LaTeX specification so I could use some. A function signature linking existing types would be a great start.

Reply:
Any help with this field or a lens and the needed type here would be very welcome.